src/main/java/seedu/address/logic/commands/SearchCommand.java

```diff
@@ -0,0 +1,229 @@
+        }
+
+        public void setEmail(Email email) {
+            this.email = email;
+        }
+
+        public Optional<Email> getEmail() {
+            return Optional.ofNullable(email);
+        }
+
+        public void setCountry(Country country) {
+            this.country = country;
+        }
+
+        public Optional<Country> getCountry() {
+            return Optional.ofNullable(country);
+        }
+
+        public void setStatus(Status status) {
+            this.status = status;
+        }
+
+        public Optional<Status> getStatus() {
+            return Optional.ofNullable(status);
+        }
+
+        public void setComment(Comment comment) {
+            this.comment = comment;
+        }
+
+        public Optional<Comment> getComment() {
+            return Optional.ofNullable(comment);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        public SearchPredicate getPredicate() {
+            ArrayList<ContainsKeywordsPredicate> predicateList = new ArrayList<>();
+            ContainsKeywordsPredicate<Name> nameSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_NAME, this.getName());
+            predicateList.add(nameSearch);
+            ContainsKeywordsPredicate<Phone> phoneSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_PHONE, this.getPhone());
+            predicateList.add(phoneSearch);
+            ContainsKeywordsPredicate<Email> emailSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_EMAIL, this.getEmail());
+            predicateList.add(emailSearch);
+            ContainsKeywordsPredicate<Country> countrySearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_COUNTRY, this.getCountry());
+            predicateList.add(countrySearch);
+            ContainsKeywordsPredicate<Status> statusSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_STATUS, this.getStatus());
+            predicateList.add(statusSearch);
+            ContainsKeywordsPredicate<Comment> commentSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_COMMENT, this.getComment());
+            predicateList.add(commentSearch);
+            ContainsKeywordsPredicate<Set<Tag>> tagSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_TAG, this.getTags());
```

Comment:
so if optional is empty, it matches every criteria thus returning everything is it?

Reply:
Yes (for that attribute): ContainsKeywordPredicate is designed in a way such that empty Optionals (for a particular attribute) will always return true for the test predicate method.